### ISSUE TITLE
Does not build: "expression is not GC-safe"

### ISSUE THREAD

**Issue by enthus1ast** [mrhdias/EnigmaHTTPServer#1]
```
$ nim --version
Nim Compiler Version 0.19.9 [Linux: amd64]
Compiled at 2019-05-08
Copyright (c) 2006-2019 by Andreas Rumpf
```

```
$ nim c server.nim

server.nim(245, 15) Error: type mismatch: got <AsyncHttpServer, Port, proc (req: Request): Future[system.void]{.locks: <unknown>.}>
but expected one of: 
proc serve(server: AsyncHttpServer; port: Port;
          callback: proc (request: Request): Future[void] {.closure, gcsafe.};
          address = ""): Future[void]
  first type mismatch at position: 3
  required type: proc (request: Request): Future[system.void]{.closure, gcsafe.}
  but expression 'handler' is of type: proc (req: Request): Future[system.void]{.locks: <unknown>.}
  This expression is not GC-safe. Annotate the proc with {.gcsafe.} to get extended error information.

expression: serve(server, Port(8080), handler)
```

**Comment by enthus1ast** [comment]
annotating the handler with gcsafe fixed it for me
`proc handler(req: Request) {.async, gcsafe.} =`

**Comment by aranhaagency** [comment]
it's working for me.
```
$ nim --version
Nim Compiler Version 0.19.4 [MacOSX: amd64]
Compiled at 2019-02-01
Copyright (c) 2006-2018 by Andreas Rumpf

**Comment by mrhdias** [comment]
This issue has been fixed with Nim version 0.20.0.
